compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/ZipUtils.java

```diff
@@ -59,38 +58,52 @@ static void generateBalo(BLangPackage bLangPackage, String projectPath, Stream<P
             try {
                 Files.createDirectories(destPath);
             } catch (IOException e) {
-                throw new BLangCompilerException("Error occurred when creating directories in " +
-                                                         "./ballerina/repo/ to save the generated balo");
+                throw new BLangCompilerException("error creating directories in ./ballerina/repo/ to save the " +
```

Comment:
This is a temp hack I did, can you please improve using createAndGetHomeReposPath()